tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -1791,6 +1791,32 @@ public static function provideIsBinaryOperator80Cases(): iterable
         ];
     }
 
+    /**
+     * @dataProvider provideIsBinaryOperator81Cases
+     * @requires PHP 8.1
+     */
+    public function testIsBinaryOperator81(array $expected, string $source): void
+    {
+        $tokensAnalyzer = new TokensAnalyzer(Tokens::fromCode($source));
+
+        foreach ($expected as $index => $isBinary) {
+            static::assertSame($isBinary, $tokensAnalyzer->isBinaryOperator($index));
+
+            if ($isBinary) {
+                static::assertFalse($tokensAnalyzer->isUnarySuccessorOperator($index));
+                static::assertFalse($tokensAnalyzer->isUnaryPredecessorOperator($index));
+            }
+        }
+    }
+
+    public static function provideIsBinaryOperator81Cases(): iterable
+    {
+        yield 'intersection is not a binary operator' => [
+            [6 => false],
+            '<?php function test(A&B $var): void {}',
```

Comment:
Can we have here very similar code like `test(A&B)` and `function test($var = A&B) {}` as in both `&` is bitwise "and"?

Reply:
Added more tests, please see last commit